Comments: avoid PHP notice when Jetpack's connection is broken.

jeherve opened the pull request:
#### Changes proposed in this Pull Request:

This should solve notices like this one:

```
E_NOTICE Trying to get property 'secret' of non-object
/wp-content/plugins/jetpack/modules/comments/comments.php:315

E_NOTICE Trying to get property 'secret' of non-object
/wp-content/plugins/jetpack/modules/comments/comments.php:333
```

#### Other information:

- [ ] Have you written new tests for your changes, if applicable?
- [ ] Have you checked the E2E test CI results, and verified that your changes do not break them?

#### Jetpack product discussion

* No

#### Does this pull request change what data or activity we track or use?

* No

#### Testing instructions:

I'm not quite sure how to reproduce the notices yet.

* You would need to start with a site that's connected to WordPress.com, and where you've activated the Comments module under Jetpack > Settings > Discussion
* Then you would have to break or delete the blog token.
* Then, on the frontend of the site when logged out, visit a post where comments are opened.
* You should see the default comment form.

jeherve committed:
Comments: avoid PHP notice when Jetpack's connection is broken.

This should solve notices like this one:

```
E_NOTICE Trying to get property 'secret' of non-object
/wp-content/plugins/jetpack/modules/comments/comments.php:315

E_NOTICE Trying to get property 'secret' of non-object
/wp-content/plugins/jetpack/modules/comments/comments.php:333
```

jeherve committed:
Update package version

sdixon194 committed:
Merge remote-tracking branch 'origin/trunk' into fix/comments-notice

sdixon194 reviewed (dismissed):
After testing a bunch and trying to break my test site a bunch of ways, I also couldn't reproduce the notice, but the change seems logical.

anomiex commented on projects/plugins/jetpack/modules/comments/comments.php:
```diff
@@ -311,7 +311,12 @@ public function comment_form_after() {
+		$blog_token = ( new Tokens() )->get_access_token();
+		// If we have no token, bail.
+		if ( ! $blog_token || is_wp_error( $blog_token ) ) {
+			return;
```
Should we output some sort of indication that there was an error, instead of having no comment form being output with no explanation?

jeherve replied:
That's a good point. What do you think about extracting this a bit, so we can bail entirely and get the default core comment form when we do not have a healthy connection to WordPress.com?
e9f60bc7a83f15736f295ce4ada4047dc212859d

anomiex replied:
Fine with me.

anomiex reviewed (commented):
Looks ok for what it does. One suggestion inline.

Also I noticed similar code at https://github.com/Automattic/jetpack/blob/a7c785206d199fc95994f44b87f4d3b464996745/projects/plugins/jetpack/modules/comments/comments.php#L560-L563
That handles getting returned false, but not the WP_Error case. Should it check for that too (maybe outputting the error)?

jeherve committed:
Handle WP_Errors

See https://github.com/Automattic/jetpack/pull/25127#pullrequestreview-1045027574

jeherve committed:
Extract blog token extraction

This allows us to bail before we even do anything with the default comment form when we do not have a healthy connection.

See https://github.com/Automattic/jetpack/pull/25127#discussion_r925605419
